### PR TITLE
OpenAPI spec

### PR DESCRIPTION
**Pull request by f-algebra**

### CODE REVIEW

File: packages/common/openapi.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+          type: string
+        title:
+          type: string
+        contactName:
+          type: string
+        requiredPeopleCount:
+            type: number
+        startTime:
+          type: number
+          example: 1666817899
```
Comment (stabback) [inline packages/common/openapi.yml:95]: Can we talk with ISO 8601 date strings?

Reason - bakes the TZ right in for easier handling.

File: packages/common/openapi.yml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+     
+    Opportunity:
+      type: object
+      allOf:
+      - $ref: "#/components/schemas/OpportunitySummary"
+      - properties:
+          description:
+            type: string
+          locationName:
+            type: object
```
Comment (stabback) [inline packages/common/openapi.yml:111]: Should this be a string?  Or is this going to be a complex object that describes an address?

File: packages/common/openapi.yml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+          idealVolunteer:
+            type: string
+          additionalInformation:
+            type: string
+
+    User:
+      type: object
+      properties:
+        userId:
+          type: string
```
Comment (stabback) [inline packages/common/openapi.yml:132]: We could probably get the name in as part of the stub at this point, we can worry about roles later.

**Review by stabback — COMMENTED** [review]
Thanks for catching the details!  We should probably bake in pagination, requiring filter params and page details.

Probably best to follow up with the admin stuff linking people to opportunities later.

Did you want to include an endpoint for submitting interest?

File: packages/common/openapi.yml (hunk 2 above)
Comment (will-sonnex) [reply packages/common/openapi.yml:111]: This was a mistake I think from an earlier draft. String makes sense.

File: packages/common/openapi.yml (hunk 1 above)
Comment (will-sonnex) [reply packages/common/openapi.yml:95]: As discussed in separate chat, we are going to go with epoch time for this project.

**Comment by will-sonnex** [comment]
Moving to auto-generating the spec from backend code

**Pull request closed without merging**